source/common/grpc/json_transcoder_filter.cc

```diff
@@ -0,0 +1,398 @@
+        request_stream_(request_translator_->Output().CreateInputStream()),
+        response_stream_(response_translator_->CreateInputStream()) {}
+
+  // Transcoder
+  ::google::grpc::transcoding::TranscoderInputStream* RequestOutput() {
+    return request_stream_.get();
+  }
+  Status RequestStatus() { return request_translator_->Output().Status(); }
+
+  ZeroCopyInputStream* ResponseOutput() { return response_stream_.get(); }
+  Status ResponseStatus() { return response_translator_->Status(); }
+
+private:
+  std::unique_ptr<JsonRequestTranslator> request_translator_;
+  std::unique_ptr<ResponseToJsonTranslator> response_translator_;
+  std::unique_ptr<TranscoderInputStream> request_stream_;
+  std::unique_ptr<TranscoderInputStream> response_stream_;
+};
+
+} // namespace
+
+JsonTranscoderConfig::JsonTranscoderConfig(const Json::Object& config) {
+  const std::string proto_descriptor_file = config.getString("proto_descriptor");
+  FileDescriptorSet descriptor_set;
+  if (!descriptor_set.ParseFromString(Filesystem::fileReadToEnd(proto_descriptor_file))) {
+    throw EnvoyException("transcoding_filter: Unable to parse proto descriptor");
+  }
+
+  for (const auto& file : descriptor_set.file()) {
+    if (descriptor_pool_.BuildFile(file) == nullptr) {
+      throw EnvoyException("transcoding_filter: Unable to build proto descriptor pool");
+    }
+  }
+
+  // TODO(lizan): Consider factor out building PathMatcher building.
+  google::grpc::transcoding::PathMatcherBuilder<const google::protobuf::MethodDescriptor*> pmb;
+
+  for (const auto& service_name : config.getStringArray("services")) {
+    auto service = descriptor_pool_.FindServiceByName(service_name);
+    if (service == nullptr) {
+      throw EnvoyException("transcoding_filter: Could not find '" + service_name +
+                           "' in the proto descriptor");
+    }
+    for (int i = 0; i < service->method_count(); ++i) {
+      auto method = service->method(i);
+
+      auto http_rule = method->options().GetExtension(google::api::http);
+
+      switch (http_rule.pattern_case()) {
+      case ::google::api::HttpRule::kGet:
+        pmb.Register("GET", http_rule.get(), http_rule.body(), method);
+        break;
+      case ::google::api::HttpRule::kPut:
+        pmb.Register("PUT", http_rule.put(), http_rule.body(), method);
+        break;
+      case ::google::api::HttpRule::kPost:
+        pmb.Register("POST", http_rule.post(), http_rule.body(), method);
+        break;
+      case ::google::api::HttpRule::kDelete:
+        pmb.Register("DELETE", http_rule.delete_(), http_rule.body(), method);
+        break;
+      case ::google::api::HttpRule::kPatch:
+        pmb.Register("PATCH", http_rule.patch(), http_rule.body(), method);
+        break;
+      case ::google::api::HttpRule::kCustom:
+        pmb.Register(http_rule.custom().kind(), http_rule.custom().path(), http_rule.body(),
+                     method);
+        break;
+      default:
+        break;
```

Comment:
The only possibility is `::google::api::HttpRule::PATTEN_NOT_SET` (added a comment here), and we do nothing here.